spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerOperations.java

```diff
@@ -0,0 +1,148 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.core;
+
+import java.util.List;
+
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.Key;
+import com.google.cloud.spanner.KeySet;
+import com.google.cloud.spanner.Options;
+import com.google.cloud.spanner.Statement;
+
+/**
+ * Defines operations available to use with Spanner.
+ *
+ * @author Ray Tsang
+ * @author Chengyuan Zhao
+ */
+public interface SpannerOperations {
+
+	/**
+	 * Gets the Spanner database client used to execute the operations.
+	 * @return the Spanner database client.
+	 */
+	DatabaseClient getDatabaseClient();
```

Comment:
I would like to remove this from the operations interface, because the idiomatic way to expose the DatabaseClient should be providing a FactoryBean for it, hence users can inject it easily if they need it. Putting this responsibility on the Template seems like mixing concerns, it's an implementation detail.

Reply:
An example to expose the DatabaseClient could be the way [MongoClient is exposed in the MongoDb module](https://github.com/spring-projects/spring-data-mongodb/blob/master/spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/MongoClientFactoryBean.java).

Reply:
@balopat done. PTAL